docs/src/language_servers.md

```diff
@@ -12,6 +12,69 @@ If there are no setup steps for a language server on this page, but a [language
 !!! info "For legacy ST3 docs, see [lsp.readthedocs.io](https://lsp.readthedocs.io)."
 
 
+## Universal
+
+### Tabby
+
+[Tabby](https://tabby.tabbyml.com/) is a self-hosted AI coding assistant which can provide inline completions for [various programming languages](https://tabby.tabbyml.com/docs/references/programming-languages/).
+
+In order to use Tabby you need a sufficiently fast GPU; the CPU version which can also be downloaded from the GitHub releases page is much too slow and it will result in timeouts for the completion requests.
+Alternatively, Tabby can be setup on a separate server with capable hardware; see the [Configuration docs](https://tabby.tabbyml.com/docs/extensions/configurations/) for the required configuration details.
+The following steps describe a local installation on a Windows PC with compatible Nvidia GPU. More installation methods and the steps for other operation systems are listed in the [Tabby docs](https://tabby.tabbyml.com/docs/quick-start/installation/docker/).
+
+1. Download and install the CUDA Toolkit from <https://developer.nvidia.com/cuda-downloads>
+
+2. Download and extract a CUDA version of Tabby from the [GitHub releases page](https://github.com/TabbyML/tabby/releases) (click on "Assets"); e.g. `tabby_x86_64-windows-msvc-cuda122.zip`
+
```

Comment:
There is an additional step here.
Extract the zip, and change the access permissions of `llama-server` and `tabby` to be executables:
```
$  tabby_aarch64-apple-darwin ls
llama-server tabby
$ tabby_aarch64-apple-darwin chmod +x llama-server
$  tabby_aarch64-apple-darwin chmod +x tabby
```

> [!NOTE] 
> On MacOS you might get an error `“tabby” cannot be opened because it is from an unidentified developer.`
> After changing the permission to executable, right click on `tabby` and select `Open`, that will get rid of the error.